pyflakes/checker.py

```diff
@@ -734,8 +734,44 @@ def NAME(self, node):
             # arguments, but these aren't dispatched through here
             raise RuntimeError("Got impossible expression context: %r" % (node.ctx,))
 
+    def CONTINUE(self, node):
+        n = node
+        while hasattr(n, 'parent'):
+            n = n.parent
+            if isinstance(n, (ast.While, ast.For)):
+                # Doesn't apply unless it's in the loop itself
+                in_else = False
+                for else_node in n.orelse:
+                    for child in ast.walk(else_node):
+                        if child == node:
+                            in_else = True
+                            break
+                    if in_else:
+                        break
+                else:
+                    return
+            if isinstance(n, (ast.FunctionDef, ast.ClassDef)):
+                break
+            # Handle Try/TryFinally difference in Python < and >= 3.3
+            if hasattr(n, 'finalbody') and isinstance(node, ast.Continue):
+                in_finally = False
+                for finally_node in n.finalbody:
+                    for child in ast.walk(finally_node):
+                        if child == node:
+                            in_finally = True
+                            break
+                    if in_finally:
+                        self.report(messages.ContinueInFinally, node)
+                        return
+        if isinstance(node, ast.Continue):
+            self.report(messages.ContinueOutsideLoop, node)
+        else:  # ast.Break
+            self.report(messages.BreakOutsideLoop, node)
```

Comment:
Coming back to this (I told you I would get stalled). It's not clear to me how to break this up. Would it be clearer if it were written recursively instead of using a while loop? I can definitely add more comments explaining the algorithm and what it's checking for.